### PR TITLE
Add support for Gear API

### PR DESCRIPTION
**Pull request by bobdoah**
I've added support for the Gear API and a corresponding CLI tool. The tools supports querying gear, gear types, linking and unlinking gear from activities.

### CODE REVIEW

File: .gitignore
Comment (abrander) [inline .gitignore:1]: This .gitignore file looks very generic. I don't think it's a good fit for this project. I would suggest simply deleting it.

File: Gear.go
Comment (abrander) [inline Gear.go:17]: Do you have strong opinions about `*string`? The rest of the library uses `string` with the zero value used for "missing strings". I would suggest that the `Gear` type does the same - or we change it all other places too.

This goes for other pointer-members too.

File: Gear.go
Comment (abrander) [inline Gear.go:22]: I would suggest `float64`.

File: Gear.go
Comment (abrander) [inline Gear.go:43]: It would be awesome to have a comment describing all public functions.

File: Gear.go
Comment (abrander) [inline Gear.go:60]: All other functions in the package have a newline before the return statement. Let's add one here too.

File: Gear.go
Comment (abrander) [inline Gear.go:91]: Function body shouldn't end in a naked newline.

File: connect/gear.go
Comment (abrander) [inline connect/gear.go:67]: These lines should probably be removed.

**Review by abrander — DISMISSED** [review]
Hey, thanks for the pull request. This is awesome! Besides a few minor issues, this is highly merge-ready.

Nice first PR! :fireworks:

**Review by abrander — CHANGES_REQUESTED** [review]
If these minor issues are resolved, we're good to merge.

File: .gitignore
Comment (bobdoah) [reply .gitignore:1]: It's generated from https://www.toptal.com/developers/gitignore/api/vim,go, where you put in the relevant languages and editors and it generates you a specific config. Half of it is specific to Golang projects. I'm not overly desperate to keep it, but I generated it to stop myself checking in VIM swap files.

File: Gear.go
Comment (bobdoah) [reply Gear.go]: I can't really recall why I did this, I should have kept more granular commit history.

**Comment by bobdoah** [comment]
> If these minor issues are resolved, we're good to merge.

I believe these should be fixed now. I noticed there were other functions with misplaced or missing newlines. I've added comments to all the public method, let me know if they're not descriptive enough.

**Review by abrander — APPROVED** [review]
This looks awesome. I'll merge right away. And then deal with the go.sum file afterwards. I'm not sure how we're supposed to deal with that.

**Comment by bobdoah** [comment]
According to [this](https://blog.golang.org/using-go-modules) `go.sum` and `go.mod` are supposed to be checked in.

> Both go.mod and go.sum should be checked into version control.

**Comment by abrander** [comment]
> According to [this](https://blog.golang.org/using-go-modules) `go.sum` and `go.mod` are supposed to be checked in.

And you're absolutely right! I was referring to the hash changing after merging, but of course, it doesn't. Brainfart on me. Carry on, nothing to see here ;-)

**Pull request merged**
